src/text-fragment-utils.js

```diff
@@ -284,27 +392,47 @@ const findText = (text) => {
   if (!text) {
     return [];
   }
-  const body = document.body;
-  const treeWalker = document.createTreeWalker(body, NodeFilter.SHOW_ELEMENT, {
-    acceptNode: (node) => {
-      if (!BLOCK_ELEMENTS.includes(node.tagName)) {
-        return NodeFilter.FILTER_REJECT;
-      }
+
+  // Accept only block elements that do not contain other block elements and
```

Comment:
Isn't it possible that a node would contain the text (but not in a child) and a block element child?

```
  <div>
    This is my matching text 
    <p> and this is an irrelevant block element </p> 
  </div>
```

Reply:
This new version should work in this case and many more!